ui/app/AppLayouts/Wallet/panels/SwapInputPanel.qml

```diff
@@ -0,0 +1,272 @@
+    }
+
+    property int swapSide: SwapInputPanel.SwapSide.Pay
+    property bool fiatInputInteractive
+    property bool loading
+
+    // output API
+    readonly property string selectedHoldingId: d.selectedHoldingId
+    readonly property double cryptoValue: amountToSendInput.cryptoValueToSendFloat
+    readonly property string cryptoValueRaw: amountToSendInput.cryptoValueToSend
+    readonly property bool cryptoValueValid: amountToSendInput.inputNumberValid
+
+    // visual properties
+    property int swapExchangeButtonWidth: 44
+    property string caption: swapSide === SwapInputPanel.SwapSide.Pay ? qsTr("Pay") : qsTr("Receive")
+
+    enum SwapSide {
+        Pay = 0,
+        Receive = 1
+    }
+
+    padding: Style.current.padding
+
+    // by design
+    implicitWidth: 492
+    implicitHeight: 131
+
+    Component.onCompleted: {
+        if (root.swapSide === SwapInputPanel.SwapSide.Pay)
+            amountToSendInput.input.forceActiveFocus()
+    }
+
+    QtObject {
+        id: d
+
+        function setSelectedHoldingId(holdingId, holdingType) {
+            let holding = SQUtils.ModelUtils.getByKey(root.processedAssetsModel, "symbol", holdingId)
+            d.selectedHoldingId = holdingId
+            d.setSelectedHolding(holding, holdingType)
+        }
+
+        function setSelectedHolding(holding, holdingType) {
+            d.selectedHoldingType = holdingType
+            d.selectedHolding = holding
+            holdingSelector.setSelectedItem(holding, holdingType)
+        }
+
+        property var selectedHolding: null
+        property var selectedHoldingType: Constants.TokenType.Unknown
+        property string selectedHoldingId
+
+        readonly property bool isSelectedHoldingValidAsset: !!selectedHolding && selectedHoldingType === Constants.TokenType.ERC20
+        readonly property double maxFiatBalance: isSelectedHoldingValidAsset ? selectedHolding.currentCurrencyBalance : 0
+        readonly property double maxCryptoBalance: isSelectedHoldingValidAsset ? selectedHolding.currentBalance : 0
+        readonly property double maxInputBalance: amountToSendInput.inputIsFiat ? maxFiatBalance : maxCryptoBalance
+        readonly property string inputSymbol: amountToSendInput.inputIsFiat ? root.currencyStore.currentCurrency :
+                                                                              !!d.selectedHolding && !!d.selectedHolding.symbol ? d.selectedHolding.symbol: ""
+        readonly property string maxInputBalanceFormatted:
+            root.currencyStore.formatCurrencyAmount(prepareForMaxSend(d.maxInputBalance, d.inputSymbol), d.inputSymbol, {noSymbol: !amountToSendInput.inputIsFiat})
+
+        function prepareForMaxSend(value, symbol) {
+            if (symbol !== Constants.ethToken) {
+                return value
+            }
+
+            return value - Math.max(0.0001, Math.min(0.01, value * 0.1))
+        }
+
+        property string searchText
+    }
+
+    background: Shape {
```

Comment:
<img width="723" alt="Screenshot 2024-06-05 at 12 40 08 PM" src="https://github.com/status-im/status-desktop/assets/60327365/b4518f93-a600-45d7-909d-614554fb8c12">

The highlight isnt looking right? let me know if you have the same?

Reply:
Can't immediately see what's incorrect with it?

Reply:
For reference, this is what I'm seeing:

![image](https://github.com/status-im/status-desktop/assets/5377645/338b4d64-c16d-44e0-82a4-220fdf316472)

Reply:
notice the semicircle on top? it doesnt look smooth?

Reply:
is the component highlighted in yours?